inc/tracking/class-tracking.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Pressbooks\Tracking;
+
+abstract class Tracking {
+	protected static $instance;
+
+	/**
+	 * Tracking table
+	 *
+	 * @var string
+	 */
+	protected $dbTable;
+
+	/**
+	 * Tracking type
+	 *
+	 * @var string
+	 */
+	protected $type;
+
+	protected function __construct() {
+		global $wpdb;
+
+		$this->dbTable = $wpdb->base_prefix . 'pressbooks_tracking';
+	}
+
+	public static function init() {
+		if ( is_null( static::$instance ) ) {
+			static::$instance = new static;
+
+			static::$instance->setup();
+		}
+
+		return static::$instance;
+	}
+
+	/**
+	 * Set up the database table.
+	 *
+	 * @return void
+	 */
+	protected function setup() {
+		require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+
+		$sql = "CREATE TABLE IF NOT EXISTS `$this->dbTable` (
+				`id` bigint(20) NOT NULL AUTO_INCREMENT,
+				`blog_id` bigint(20) NOT NULL,
+				`track_type` varchar(30) NOT NULL,
+				`track_value` varchar(255),
```

Comment:
I wonder if we want to store serialized data in the future for another tracking, makes more sense create this field as a `text` type. We are not sure at the moment, so keep it as varchar(255) is ok to me.

Reply:
Like you said, I think we can keep that as `varchar` for now and possibly change to `text` as needed.